High Sierra is out for OSX - Test and Update docs to "Sierra and higher"

## What happened (or feature request):

We shortsightedly put a [requirement for OSX Sierra](https://github.com/drud/ddev#system-requirements) in our README (probably in docs also?). So:

1. Test on High Sierra
2. Update docs to say "Sierra and higher". 

Related, #398 where the discussion is about versions of OSX prior to Sierra.

@rfay Thanks! I'll file a separate issue because that already hits me up against things like: "pkg/appimport/appimport.go:12:2: cannot find package "github.com/mitchellh/go-homedir" in any of:... make: *** [testpkg] Error 1"

--- a/docs/index.md
+++ b/docs/index.md
@@ -8,7 +8,7 @@
 
 - [Docker](https://www.docker.com/community-edition) version 17.05 or greater
 - OS Support
-  - macOS Sierra
+  - macOS Sierra and higher (macOS 10.12 and higher)
   - Linux (See [Linux notes](users/linux_notes.md))
     * Ubuntu 16.04 LTS
     * Debian Jessie

